pkg/chunk/disk_cache.go

```diff
@@ -895,7 +895,7 @@ func (cache *cacheStore) scanCached() {
 	cache.Unlock()
 }
 
-var pathReg, _ = regexp.Compile(`^chunks/\d+/\d+/\d+_\d+_\d+$`)
+var pathReg, _ = regexp.Compile(`^chunks/((\d+)|(..))/\d+/\d+_\d+_\d+$`)
```

Comment:
maybe the regex could match more precisely, like [0-9a-fA-F]{2} for hash prefix.